nestkernel/connection_manager.cpp

```diff
@@ -549,6 +549,163 @@ nest::ConnectionManager::connect( const index snode_id,
+
+  // Increments pointers to weight and delay, if they are specified.
+  auto increment_wd = [weights, delays]( decltype( weights ) & w, decltype( delays ) & d )
+  {
+    if ( weights != nullptr )
+    {
+      ++w;
+    }
+    if ( delays != nullptr )
+    {
+      ++d;
+    }
+  };
+
+  // Vector for storing exceptions raised by threads.
+  std::vector< std::shared_ptr< WrappedThreadException > > exceptions_raised( kernel().vp_manager.get_num_threads() );
+
+#pragma omp parallel
+  {
+    const auto tid = kernel().vp_manager.get_thread_id();
+    try
+    {
+      auto s = sources;
+      auto t = targets;
+      auto w = weights;
+      auto d = delays;
+      double weight_buffer = numerics::nan;
+      double delay_buffer = numerics::nan;
+      int index_counter = 0;
+
+      for ( ; s != sources + n; ++s, ++t, ++index_counter )
+      {
+        if ( 0 >= *s or static_cast< index >( *s ) > kernel().node_manager.size() )
+        {
+          throw UnknownNode( *s );
+        }
+        if ( 0 >= *t or static_cast< index >( *t ) > kernel().node_manager.size() )
+        {
+          throw UnknownNode( *t );
+        }
+        auto target_node = kernel().node_manager.get_node_or_proxy( *t, tid );
+        if ( target_node->is_proxy() )
+        {
+          increment_wd( w, d );
+          continue;
+        }
+
+        // If weights or delays are specified, the buffers are replaced with the values.
+        // If not, the buffers will be NaN and replaced by a default value by the connect function.
+        if ( weights != nullptr )
+        {
+          weight_buffer = *w;
+        }
+        if ( delays != nullptr )
+        {
+          delay_buffer = *d;
+        }
+
+        // Store the key-value pair of each parameter in the Dictionary.
+        for ( auto& param_pointer_pair : param_pointers )
+        {
+          // Increment the pointer to the parameter value.
+          auto* param = param_pointer_pair.second + index_counter;
+
+          // Receptor type must be an integer.
+          if ( param_pointer_pair.first == names::receptor_type )
+          {
+            const auto rtype_as_long = static_cast< long >( *param );
+
+            if ( *param > 1L << 31 or std::abs( *param - rtype_as_long ) > 0 ) // To avoid rounding errors
+            {
+              throw BadParameter( "Receptor types must be integers." );
+            }
+
+            // Change value of dictionary entry without allocating new datum.
+            auto id = static_cast< IntegerDatum* >( ( ( *param_dicts[ tid ] )[ param_pointer_pair.first ] ).datum() );
+            ( *id ) = rtype_as_long;
+          }
+          else
+          {
+            auto dd = static_cast< DoubleDatum* >( ( ( *param_dicts[ tid ] )[ param_pointer_pair.first ] ).datum() );
+            ( *dd ) = *param;
+          }
+        }
+
+        connect(
+          *s, target_node, tid, synapse_model_id, param_dicts[ tid ], delay_buffer, weight_buffer );
```

Comment:
I think this is what is causing the problems on Travis, it should be one line.

Reply:
Thanks, this seems to have solved the issue!